### ISSUE TITLE
StatefulSet vs Deployment

### ISSUE THREAD

**Issue by eric-pierre** [gadsme/charts#44]
Hi!

I will confess being completly new to CubeJs, which is why my question may sound profoundly ignorant, but I was wondering whether the worker and router pods could be scheduled via a Deployment instead of a StatefulSet? What would be the consequence?

Thanks for your kind help!

**Comment by lvauvillier** [comment]
Hi @eric-pierre,

We require a StatefulSet because we deploy a headless service, allowing worker instances to receive stable names to communicate each other and declared to the router.

More info here:
https://kubernetes.io/docs/concepts/services-networking/service/#headless-services

Router registration:
https://github.com/gadsme/charts/blob/main/charts/cubestore/templates/router/statefulset.yaml#L83

Worker registration:
https://github.com/gadsme/charts/blob/main/charts/cubestore/templates/workers/statefulset.yaml#L70

**Comment by eric-pierre** [comment]
Thank you so much for your kind elaboration- Just as a follow up- Could we
use a regular service instead of headless? Again I apologize if this is a
silly question!

**Comment by lvauvillier** [comment]
Yes, you can if you write your own files and manually set the `CUBESTORE_WORKERS`env var.

But in the context of this chart this is the only way i found to properly generate list of stable pod dns names from a simple `workers.workersCount` config.

**Comment by eric-pierre** [comment]
Thanks! This helps a lot! If I were to use a regular service, I could
essentially configure the router to point to it, and let the LB worry about
the actual routing. I think that would work, unless the router needs to
know about the individual worker instances- This is where my knowledge of
CubeJS fails me.

**Comment by lvauvillier** [comment]
I updated my previous comment. When using a load balancer, the CUBESTORE_WORKERS value becomes a single entry point. However, CubeStore requires direct access to each individual pod.
